ksqldb-engine/src/main/java/io/confluent/ksql/planner/plan/UserRepartitionNode.java

```diff
@@ -103,9 +107,19 @@ public Expression resolveSelect(final int idx, final Expression expression) {
 
   @Override
   void validateKeyPresent(final SourceName sinkName, final Projection projection) {
-    if (!(partitionBy instanceof NullLiteral) && !projection.containsExpression(partitionBy)) {
-      final ImmutableList<Expression> keys = ImmutableList.of(originalPartitionBy);
+    if (!PartitionByParamsFactory.isPartitionByNull(partitionBy)
+        && !containsExpressions(projection, partitionBy)) {
+      final ImmutableList<Expression> keys = ImmutableList.copyOf(originalPartitionBy);
```

Comment:
do we still need to make a copy of the original partition by now that it's a list?

Reply:
Nope, removed.